medallion/test/test_backends.py

```diff
@@ -1503,3 +1386,62 @@ def test_default_backend_no_backend_section(no_backend_section):
     assert no_backend_section.app.medallion_backend.data == {}
 
 # test collections with different can_read and can_write values
+
+
+# test if program will reject duplicate objects being posted
+def test_object_already_present(backend):
+    object_copy = {
+                        "created": "2014-05-08T09:00:00.000Z",
+                        "modified": "2014-05-08T09:00:00.000Z",
+                        "id": "relationship--2f9a9aa9-108a-4333-83e2-4fb25add0463",
+                        "relationship_type": "indicates",
+                        "source_ref": "indicator--cd981c25-8042-4166-8945-51178443bdac",
+                        "spec_version": "2.1",
+                        "target_ref": "malware--c0931cc6-c75e-47e5-9036-78fabc95d4ec",
+                        "type": "relationship"
+                    }
+    object_copy2 = object_copy.copy()
+    del object_copy2['modified']
+    add_objects = {"objects": []}
+
+    add_objects["objects"].append(object_copy)
+    # add object to test against
+    r_post = backend.client.post(
+        test.ADD_OBJECTS_EP,
+        data=json.dumps(add_objects),
+        headers=backend.post_headers,
+    )
+
+    add_objects["objects"].append(object_copy2)
+    # try to add a duplicate, with and without the modified key (both should fail)
+    r_post = backend.client.post(
+        test.ADD_OBJECTS_EP,
+        data=json.dumps(add_objects),
+        headers=backend.post_headers,
+    )
+    status_data = r_post.json
+    assert r_post.status_code == 202
+    # should have failures
+    assert "failures" in status_data
+    # should not have successes
+    assert "successes" not in status_data
+
+
+def test_save_to_file(backend):
+    if backend.type != "memory":
+        pytest.skip()
+    with tempfile.NamedTemporaryFile(mode='w') as tmpfile:
+        backend.app.medallion_backend.save_data_to_file(tmpfile.name)
+        tmpfile.flush()
+        with open(tmpfile.name) as f:
+            data = json.load(f)
+        if 'trustgroup1' in data.keys():
+            if 'collections' in data['trustgroup1'].keys():
+                assert data['trustgroup1']['collections'][0]['id'] == "472c94ae-3113-4e3e-a4dd-a9f4ac7471d4"
+                assert data['trustgroup1']['collections'][1]['id'] == "365fed99-08fa-fdcd-a1b3-fb247eb41d01"
+                assert data['trustgroup1']['collections'][2]['id'] == "91a7b528-80eb-42ed-a74d-c6fbd5a26116"
+                assert data['trustgroup1']['collections'][3]['id'] == "52892447-4d7e-4f70-b94d-d7f22742ff63"
+            else:
+                assert False
+        else:
+            assert False
```

Comment:
In tests we don't really need to handle the case where a key isn't present in a dictionary as a special case. If the key's not there, something went wrong, an error will raise, and the test will fail. No need to assert False.

```suggestion
        assert data['trustgroup1']['collections'][0]['id'] == "472c94ae-3113-4e3e-a4dd-a9f4ac7471d4"
        assert data['trustgroup1']['collections'][1]['id'] == "365fed99-08fa-fdcd-a1b3-fb247eb41d01"
        assert data['trustgroup1']['collections'][2]['id'] == "91a7b528-80eb-42ed-a74d-c6fbd5a26116"
        assert data['trustgroup1']['collections'][3]['id'] == "52892447-4d7e-4f70-b94d-d7f22742ff63"
```